### PR TITLE
1341-MoveAlmaTransformationToFix

### PR DESCRIPTION
**Pull request by TobiasNx**
Transform the describedBy morph into fix. Also move all other transformation to fix.

See #1341.

### CODE REVIEW

File: src/main/resources/alma/fix/subjects.fix [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+        copy_field("$j", "subject[].$last.componentList[].$last.gndIdentifier")
+        replace_all("subject[].$last.componentList[].$last..gndIdentifier","^\\(DE-588\\)(.*)$","$1") 
+      end
+    end
+  end
+end
+
+# TODO: Find out why this does not work?
+# lookup("subject[].*.componentList[].*.type[].*","rswk-indicator")
+
```
Comment (TobiasNx) [inline src/main/resources/alma/fix/subjects.fix:389]: @blackwinter do you have an idea why this breaks?

Comment (TobiasNx) [reply src/main/resources/alma/fix/subjects.fix:389]:
```
Caused by: java.lang.IllegalArgumentException: Using ref, but can't find: 2 in: null
        at org.metafacture.metafix.FixPath.insertInto(FixPath.java:298)
        at org.metafacture.metafix.FixPath.insertInto(FixPath.java:265)
        at org.metafacture.metafix.FixPath.lambda$insertInto$20(FixPath.java:293)
        at org.metafacture.metafix.Value$TypeMatcher.match(Value.java:344)
        at org.metafacture.metafix.Value$TypeMatcher.ifArray(Value.java:317)
        at org.metafacture.metafix.FixPath.lambda$insertInto$22(FixPath.java:293)
```

Comment (TobiasNx) [reply src/main/resources/alma/fix/subjects.fix:599]: It runs with: `$ mvn clean install -DskipTests=true ; mvn -DgenerateTestData=true failsafe:integration-test -Dit.test=AlmaMarc21XmlToLobidJsonMetafixTest.java`

Comment (fsteeg) [reply src/main/resources/alma/fix/subjects.fix:618]: I'm afraid I have no quick fix here, it's actually surprising how that current path update logic works at all, I'll keep trying.

Comment (TobiasNx) [reply src/main/resources/alma/fix/subjects.fix]: Thanks this seems to work now!

**Comment by TobiasNx** [comment]
Okay, this seems like a very huge branch and MR.
But besides the rest of TODOs the main construction for the new transformation seems done.

While changing the ALMA-transformation to fix I stumbled over some problems for that I needed to go back to the old ALEPH Morph especially with the `medium` and `type` elements. Here are some questions open concerning the mappings and the possibilities it seems that some info from Aleph is lost "forever". Others seem to have no proper mappings.

Things I suggest to do now is create an index from the new fix transformation. Repair all broken things with real data. Then start finishing and cleaning.

**Comment by TobiasNx** [comment]
Added `hasItem` POR transformation also changed `oclcNumber` to array instead of `string`

Note to @hagbeck : when we move the ALMA Transformation to Fix, oclcNumber will be an array. Should we already implement this in the old ALMA-Morph, or is this change not relevant to you?

**Comment by hagbeck** [comment]
@TobiasNx We are using the `oclcNumber` but the changes are not relevant for us now. So no changing of the ALMA-Morph is necessary.

**Comment by dr0i** [comment]
Merged into master. We will go on using smaller PRs to solve expected issues. Closed.

**Pull request merged**